product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCase.java

```diff
@@ -26,48 +26,92 @@
     private APIPublisherRestClient apiPublisherAdmin;
+    private APIStoreRestClient apiStore;
+
     private static final String ADMIN_LOGIN_USERNAME = "admin";
     private static final String ADMIN_LOGIN_PW = "admin";
+    private static final String LOGIN_PERMISSION = "/permission/admin/login";
     private static final String API_ADMIN_PERMISSION = "/permission/admin";
-    private static final String API_PUBLISHER_PERMISSION = "/permission/admin/manage/api/publish";
-    private static final String API_CREATOR_PERMISSION = "/permission/admin/manage/api/create";
+    private static final String API_SUBSCRIBER_PERMISSION = "/permission/admin/manage/api/subscribe";
     private static final String MANAGER_ROLE = "managerRole";
     private static final String AGENT_ROLE = "agentRole";
+    private static final String AGENT_LOGIN_PW = "agent";
     private static final String SUPER_USER = "Harry";
     private static final String SUPER_USER_LOGIN_PW = "super";
+    private static final String CUSTOMER_ROLE = "customerRole";
+    private static final String CUSTOMER_LOGIN_PW = "customer";
     private static final String ITEM_VIEW = "item_view";
     private static final String ITEM_ADD = "item_add";
     private static final String ORDER_VIEW = "order_view";
     private static final String ORDER_ADD = "order_add";
     private static final String SCOPE_EXISTANCE = "isScopeExist";
     private static final String ROLE_EXISTANCE = "isRoleExist";
+    private static final String AGENT = "Paul";
+    private static final String CUSTOMER = "Richard";
+
+
     List<String> userList = new ArrayList();
     List<String> roleList = new ArrayList();
     private String apiVersion = "1.0.0";
     private String apiVisibility = "public";
     private String backendEndPoint = "http://ws.cdyne.com/phoneverify/phoneverify.asmx";
     private String apiName = "APIScopeTestAPI";
+    private String applicationName = "TestApplication3";
     private File swaggerFile;
+    JSONObject swaggerJson;
     String resourceLocation = System.getProperty("test.resource.location");
 
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        setupUserData();
+        apiPublisher = new APIPublisherRestClient(publisherURL);
+        apiPublisher.login(SUPER_USER, SUPER_USER_LOGIN_PW);
+        apiPublisherAdmin = new APIPublisherRestClient(publisherURL);
+        apiPublisherAdmin.login(ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+        apiStore = new APIStoreRestClient(storeURL);
+
+        // create and publish sample API
+        String swaggerFilePath = resourceLocation + "swaggerFiles" + File.separator + "APIScopeTest2.json";
+        File swaggerFile = new File(swaggerFilePath);
+        String swaggerContent = readFromFile(swaggerFile.getAbsolutePath());
+        swaggerJson = new JSONObject(swaggerContent);
+
+        try {
+            apiPublisher.developSampleAPI(swaggerJson, SUPER_USER, backendEndPoint, true, apiVisibility);
```

Comment:
Better throw this exception without catching, since subsequent tests will anyway fail due to this